agent/config/config.go

```diff
@@ -297,6 +304,25 @@ func environmentConfig() Config {
 		seelog.Warnf("Invalid format for \"ECS_NUM_IMAGES_DELETE_PER_CYCLE\", expected an integer. err %v", err)
 	}
 
+	instanceAttributesEnv := os.Getenv("ECS_INSTANCE_ATTRIBUTES")
+	attributeDecoder := json.NewDecoder(strings.NewReader(instanceAttributesEnv))
+	var instanceAttributes map[string]string
+
+	err = attributeDecoder.Decode(&instanceAttributes)
+	if err != io.EOF && err != nil {
+		msg := "Invalid format for ECS_INSTANCE_ATTRIBUTES. Expected a json hash"
```

Comment:
You're not just expecting a JSON hash, but rather a `map[string]string` (or a `Map<String, String>` if you prefer Java notation).  If someone has the following valid JSON it will still be broken:

```json
{
	"field": 1,
	"field2": "string"
}
```

(and I think in that case you may just get a silent failure)

You might want to do something like:

```go
var decodedInstanceAttributes map[string]interface{}
// error check decoding
instanceAttributes := map[string]string{}
for decodedAttribute, decodedValue := range decodedInstanceAttributes {
	value, ok := decodedValue.(string)
	if ok {
		instanceAttributes[decodedAttribute] = value
	} else {
		// found an invalid attribute; value was not a string
		// warn here
	}
}